### PR TITLE
File size limit + ignore empty messages.

### PR DESCRIPTION
**Pull request by keesschepers**

### CODE REVIEW

File: PersistenceHandler/RabbitMqPersistenceHandler.php [hunk 1]
```diff
@@ -26,15 +27,22 @@ class RabbitMqPersistenceHandler implements PersistenceHandler
...
+     */
+    private $maximumResourceSize;
+
+    /**
      * Constructor.
      *
-     * @param \Simgroep\ConcurrentSpiderBundle\Queue $queue
+     * @param \Simgroep\ConcurrentSpiderBundle\Queue                             $queue
      * @param \Simgroep\ConcurrentSpiderBundle\DocumentResolver\DocumentResolver $documentResolver
+     * @param integer                                                            $maximumResourceSize
```
Comment (lkalinka) [inline PersistenceHandler/RabbitMqPersistenceHandler.php]: its a string, not integer 16mb f.e. - i suggested by this line ->scalarNode('maximum_resource_size')->cannotBeEmpty()->defaultValue('16mb')->end() ;)

File: PersistenceHandler/RabbitMqPersistenceHandler.php [hunk 2]
```diff
@@ -59,4 +71,31 @@ public function persist(Resource $resource, CrawlJob $crawlJob)
             $this->queue->publish($message);
         }
     }
+
+    /**
+     * Function that returns the human readable size in bytes.
+     *
+     * @param string $fileSize
+     *
+     * @return integer
```
Comment (lkalinka) [inline PersistenceHandler/RabbitMqPersistenceHandler.php:80]: as above

File: PersistenceHandler/RabbitMqPersistenceHandler.php [hunk 3]
```diff
@@ -59,4 +71,31 @@ public function persist(Resource $resource, CrawlJob $crawlJob)
...
+     */
+    public static function convertToBytes($fileSize)
+    {
+        $number = substr($fileSize, 0, -2);
+
+        switch (strtoupper(substr($fileSize,-2))) {
+            case "KB":
+                return $number*1024;
```
Comment (lkalinka) [inline PersistenceHandler/RabbitMqPersistenceHandler.php]: spaces around *

File: PersistenceHandler/RabbitMqPersistenceHandler.php (hunk 2 above)
Comment (keesschepers) [inline PersistenceHandler/RabbitMqPersistenceHandler.php:80]: Not thru, the function accepts an string but returns an integer.

Comment (lkalinka) [inline PersistenceHandler/RabbitMqPersistenceHandler.php:80]: yes my mistake :P

**Pull request merged**